openedx/core/djangoapps/content/learning_sequences/models.py

```diff
@@ -67,6 +67,13 @@ class Meta:
         ]
 
 
+class CourseContext(LearningContext):
```

Comment:
Please make `CourseContext` be a model that has a foreign key to `LearningContext` instead of subclassing it.

I'd like to be able to explicitly separate things that are common to all Sequences from things that are specific to Sequences that are in a Course.  The intended relationship between models are:

* All CourseContexts are LearningContexts, but not all LearningContexts are CourseContexts (e.g. more freeform LabXchange sequences, content libraries). In terms of implementation, every row in `CourseContext` would have a unique foreign key to `LearningContext` + whatever information is specific to Courses.
* `LearningSequences` belong to a `LearningContext`
* I think you're right that `CourseSection` and `CourseSectionSequence` should move over their foreign keys to point to the new `CourseContext` model instead of `LearningContext`

So basically, we're going for two distinct sets of model relationships. There is the base set of `LearningSequence` / `LearningContext` which could apply to any sort of learning material. Different systems might want to make foreign keys to this later, like the Completion API. A student can complete any sort of Sequence, regardless of whether it's in a Course or not. The Completion API wouldn't care whether a Sequence is in a Course, Pathway, or something else we haven't thought of yet.

The second set of model relationships are Course-specific. Course visibility preferences only make sense in this context. The ordering of Sequences in a course only makes sense here. Things that operate specifically on courses would tie their models to this table.

Does that make sense?

Reply:
Yes, :blush: